### PR TITLE
DPS+Pnp implementation

### PR DESCRIPTION
**Pull request by jasmineymlo**

### CODE REVIEW

File: firmware/src/app.c [hunk 1]
```diff
@@ -114,12 +115,14 @@ static void APP_ConnectNotifyCb(DRV_HANDLE handle, WDRV_WINC_CONN_STATE currentS
...
 // *****************************************************************************
+#define SN_STRING "sn"
 
 /* Place holder for ECC608A unique serial id */
 ATCA_STATUS appCryptoClientSerialNumber;
 ATCA_STATUS retValCryptoClientSerialNumber;
 char* attDeviceID;
-char attDeviceID_buf[20] = "BAAAAADD1DBAAADD1D";
+char attDeviceID_buf[25] = "BAAAAADD1DBAAADD1D";
+char buf[25];
```
Comment (ericwolz) [inline firmware/src/app.c]: This is not big enough, you also copy "sn" into it. line 415

Comment (ericwolz) [reply firmware/src/app.c]: you should not call this buf, because it is assigned to attDeviceID

File: firmware/src/app.c [hunk 2]
```diff
@@ -337,13 +409,18 @@ void APP_Tasks(void)
             }
             else
             {
-                attDeviceID = attDeviceID_buf;
+                // To use Azure provisioning service, attDeviceID should match with the device cert CN,
+                // which is the serial number of ECC608 prefixed with "sn" if you are using the 
+                // the microchip provisioning tool for PIC24.
+                strcat(buf, SN_STRING);
```
Comment (ericwolz) [inline firmware/src/app.c]: strcpy

File: firmware/src/app.c [hunk 3]
```diff
@@ -337,13 +409,18 @@ void APP_Tasks(void)
...
+                attDeviceID = buf;
+                debug_printInfo("CRYPTO_CLIENT_printSerialNumber %s\n", attDeviceID);
             }
-#endif             
+#endif
         #if CFG_ENABLE_CLI   
             set_deviceId(attDeviceID);
-        #endif  
-            debug_init(attDeviceID); 
+        #endif
```
Comment (ericwolz) [inline firmware/src/app.c]: format

File: firmware/src/app.c [hunk 4]
```diff
@@ -465,6 +541,509 @@ static void APP_DataTask(void)
...
+    if (az_result_failed(
+            rc = az_iot_hub_client_methods_response_get_publish_topic(
+                &hub_client,
+                request->request_id,
+                status,
+                commands_response_topic,
+                sizeof(commands_response_topic),
+                NULL)))
+    {
+      debug_printError("Unable to get twin document publish topic");
```
Comment (ericwolz) [inline firmware/src/app.c]:
```suggestion
      debug_printError("Unable to get command response publish topic");
```

**Review by ericwolz — APPROVED** [review]

File: firmware/src/app.c (hunk 1 above)
Comment (jasmineymlo) [reply firmware/src/app.c]: The serial number is 18 characters long, including "sn" should be ok with 25.

File: firmware/src/app.c [hunk 5]
```diff
@@ -122,7 +122,6 @@ ATCA_STATUS appCryptoClientSerialNumber;
 ATCA_STATUS retValCryptoClientSerialNumber;
 char* attDeviceID;
 char attDeviceID_buf[25] = "BAAAAADD1DBAAADD1D";
```
Comment (ericwolz) [inline firmware/src/app.c:124]:
```suggestion
char attDeviceID_buf[25] = "snBAAAAADD1DBAAADD1D";
```

**Review by ericwolz — APPROVED** [review]

**Pull request merged**